### DISCUSSION TITLE
UI is coming along nicely!

### DISCUSSION THREAD

**Discussion by davepl** [PlummersSoftwareLLC/NightDriverStrip#487] [General]
I just updated my fork and wanted to say that the UI is coming along really nicely!  Really appreciate what folks are doing on this!

Cheers,
Dave

**Comment by KeiranHines** [comment]
Thanks Dave, 

Is there any other ideas you, or anyone else, had for the UI going forward? At this point I believe I have closed off most if not all of the pending feature requests except for the WebSocket support. 

I did consider building a "supervisor" style Application. Basically, take the existing  UI make it a standalone webserver not on-device and have the ability to connect and toggle between all your NDS devices from one webpage. I just don't know how many people would want or use it. It would really only be applicable to people that have somewhere they can run the webserver and either have multiple devices, or a device they can't run the webUI on currently but can still run the API layer.

**Comment by rbergen** [comment]
@KeiranHines Yes, you're indeed close to closing the last "feature ticket" for the web UI, bar the WebSocket one. Good job!

There is still the thing about the effect interval now being specified and shown in milliseconds, first mentioned by @robertlipe in #411. I think it makes more sense to configure and show that in seconds, which is mainly a back-end thing - which I can do myself. However, once we change the duration granularity from ms to s, the countdown should start occurring in fractions (I think tens would be granular enough?) of seconds.

Also, as also mentioned earlier, it would be nice if the interval editor on the main page would respond to Enter being pressed. Having to apply a new value by clicking somewhere outside of the field will always feel a bit odd.

**Reply by KeiranHines** [reply]
That's easy enough to do. I can do the press enter soon. For the config units I can handle all converting in the browser and the backend can stay in ms. I don't mind either way.

**Reply by rbergen** [reply]
You read my mind. It would be easier to keep the backend duration in ms - which means the field in the device config panel would also still set the time in ms. I just have to make sure in the backend that the smallest value above 0 that can actually be set is 1000 (i.e., 1 second) to prevent the board from tripping over itself.
